### PR TITLE
Add azure deploy button, add azure custom deploy.cmd script

### LINKED ISSUE #11: Deploy to Azure button
https://azure.microsoft.com/en-gb/blog/deploy-to-azure-button-for-azure-websites-2/

### PR DESCRIPTION
**Pull request by aeldar**
Fixes #11

### CODE REVIEW

File: .deployment [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[config]
```
Comment (trioletas) [inline .deployment]: we need to find a way to re-use [Kudu Deployment hook](https://github.com/ctco-dev/azure-create-react-app-kudu-deploy-hook) I did for CRA apps. Among other things (DRY), this implementation is missing a proper `web.config`, as implemented in our custom hook.

Either merge it in to this repo or somehow link the two.

Comment (aeldar) [inline .deployment]: - Create feature section in readme
- copy documentation from https://github.com/ctco-dev/azure-create-react-app-kudu-deploy-hook
- move web.config into public
- apply 12 FA changes into webpack prod config.

Comment (aeldar) [inline .deployment]: - add show case for process.env usage (show default message like (please add env variable to show actual message)

Comment (aeldar) [inline .deployment]: - put `azure` into `cloud`

Comment (aeldar) [inline .deployment]: @trioletas, updated, please review when have time for that.

File: azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
+{
```
Comment (trioletas) [inline azuredeploy.json]: Please also disable [ARR Affinity Cookie](https://blogs.msdn.microsoft.com/appserviceteam/2016/05/16/disable-session-affinity-cookie-arr-cookie-for-azure-web-apps/)

File: cloud/azure/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+Yes! The deployment script is optimized to work with Create React App but doesn't necessarily force you to use it. It just makes the assumptions about your folder structure which are valid for all CRA apps. 
+
+To deploy an arbitrary Web App with it, just make sure that
+
+- your production npm build script can be executed with the `npm run build` command
+- the distribution files created during the build are placed in the `build\` folder
+- all your static assets are inside the `build\static\` folder
+
+### Example apps
```
Comment (trioletas) [inline cloud/azure/README.md]: This is no longer required, please remove

**Comment by trioletas** [comment]
@aeldar one thing for forgot to add is no-cache policy for `index.html` file, see [web.config](https://github.com/ctco-dev/azure-create-react-app-kudu-deploy-hook/blob/master/azure/web.config#L5) from kudu hook
since `node` is now handling all index calls we have to set the header manually
`Cache-Control:no-cache` should suffice

**Comment by trioletas** [comment]
Other than that we're good to merge

**Comment by aeldar** [comment]
added `no-cache` to header for index.html

**Comment by trioletas** [comment]
thanks, looks good!

**Pull request merged**